### PR TITLE
Feature/6.redux integration

### PR DESCRIPTION
**Pull request by EmOlsz**
added redux boilerplate, login, main and 404 pages, methods for login and logout

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+    <HashRouter>
+      <Switch>
+        <Route exact path="/" component={MainPage} />
+        <Route path='/login' component={LogInPage} />
+      </Switch>
+    </HashRouter>
+  </Provider>
 );
 
 export default App;
```
Comment (kappa3-3) [inline src/components/App.jsx]: ERROR in ./src/components/App.jsx
Module not found: Error: Can't resolve 'react-router-dom' in '/home/kj/Documents/Todo-list/src/components'
 @ ./src/components/App.jsx 2:0-71 15:25-35 15:63-69 15:97-102 19:26-31 22:26-31 26:26-34
 @ ./src/index.js

File: src/pages/Page404/index.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export default function Page404() {
+  return (
+    <div>
+      <h1>Page not found :(</h1>
+      <div>
+        Go back to <Link to="/">Main page</Link>
+      </div>
+    </div>
+  )
+}
```
Comment (kappa3-3) [inline src/pages/Page404/index.jsx]: ERROR in ./src/pages/Page404/index.jsx
Module not found: Error: Can't resolve 'react-router-dom' in '/home/kj/Documents/Todo-list/src/pages/Page404'
 @ ./src/pages/Page404/index.jsx 2:0-40 4:164-168
 @ ./src/components/App.jsx
 @ ./src/index.js

File: src/components/App.jsx (hunk 1 above)
Comment (EmOlsz) [reply src/components/App.jsx]: you need to run "npm install" in console - I've added new dependency to package.json, but you have to install it manually each time this file is updated. Same goes for me, when you will add some new package and then push your changes to remote

File: src/pages/Page404/index.jsx (hunk 2 above)
Comment (EmOlsz) [reply src/pages/Page404/index.jsx]: you need to run "npm install" in console - I've added new dependency to package.json, but you have to install it manually each time this file is updated. Same goes for me, when you will add some new package and then push your changes to remote

**Comment by EmOlsz** [comment]
> Why Routing in App not external file?

Usually it's made this way - App component contains main Routing. Eventually, we can separate it as AppRouting component or something like that, and App component will just render it.
Your call

**Pull request merged**